providers/apis/binance/api_handler.go

```diff
@@ -10,115 +10,125 @@ import (
+		RightBracket,
+	), nil
 }
 
+// ParseResponse parses the response from the Binance API and returns a GetResponse. Each
+// of the tickers supplied will get a response or an error.
 func (h *APIHandler) ParseResponse(
-	cps []slinkytypes.CurrencyPair,
+	tickers []mmtypes.Ticker,
 	resp *http.Response,
-) providertypes.GetResponse[slinkytypes.CurrencyPair, *big.Int] {
+) providertypes.GetResponse[mmtypes.Ticker, *big.Int] {
 	// Parse the response into a BinanceResponse.
 	result, err := Decode(resp)
 	if err != nil {
-		return providertypes.NewGetResponseWithErr[slinkytypes.CurrencyPair, *big.Int](cps, err)
+		return providertypes.NewGetResponseWithErr[mmtypes.Ticker, *big.Int](tickers, err)
 	}
 
 	var (
-		resolved   = make(map[slinkytypes.CurrencyPair]providertypes.Result[*big.Int])
-		unresolved = make(map[slinkytypes.CurrencyPair]error)
+		resolved   = make(map[mmtypes.Ticker]providertypes.Result[*big.Int])
+		unresolved = make(map[mmtypes.Ticker]error)
 	)
 
-	// Determine of the provided currency pairs which are supported by the Binance API.
-	configuredCps := config.NewMarketConfig()
-	for _, cp := range cps {
-		market, ok := h.cfg.Market.CurrencyPairToMarketConfigs[cp.String()]
-		if !ok {
-			continue
-		}
-
-		configuredCps.CurrencyPairToMarketConfigs[cp.String()] = market
-	}
-
-	// Filter out the responses that are not expected.
+	inverted := h.market.Invert()
 	for _, data := range result {
-		market, ok := h.cfg.Market.TickerToMarketConfigs[data.Symbol]
+		// Filter out the responses that are not expected.
+		market, ok := inverted[data.Symbol]
 		if !ok {
 			continue
 		}
 
-		cp := market.CurrencyPair
-		price, err := math.Float64StringToBigInt(data.Price, cp.Decimals())
+		price, err := math.Float64StringToBigInt(data.Price, market.Ticker.Decimals)
 		if err != nil {
-			return providertypes.NewGetResponseWithErr[slinkytypes.CurrencyPair, *big.Int](cps, err)
+			unresolved[market.Ticker] = fmt.Errorf("failed to convert price %s to big.Int: %w", data.Price, err)
+			continue
 		}
 
-		resolved[cp] = providertypes.NewResult[*big.Int](price, time.Now())
-		delete(configuredCps.CurrencyPairToMarketConfigs, cp.String())
+		resolved[market.Ticker] = providertypes.NewResult[*big.Int](price, time.Now())
 	}
 
-	// If there are any currency pairs that were not resolved, return an error.
-	for _, market := range configuredCps.CurrencyPairToMarketConfigs {
-		cp := market.CurrencyPair
-		unresolved[cp] = fmt.Errorf("currency pair %s did not get a response", cp.String())
+	// Add currency pairs that received no response to the unresolved map.
+	for _, ticker := range tickers {
+		_, resolvedOk := resolved[ticker]
+		_, unresolvedOk := unresolved[ticker]
+
+		if !resolvedOk && !unresolvedOk {
+			unresolved[ticker] = fmt.Errorf("no response")
+		}
 	}
 
-	return providertypes.NewGetResponse[slinkytypes.CurrencyPair, *big.Int](resolved, unresolved)
+	return providertypes.NewGetResponse[mmtypes.Ticker, *big.Int](resolved, unresolved)
```

Comment:
Could also be `constants.NewPriceGetResponse`

Reply:
also idk if `constants` is a good package name for these type aliases.

I think `constants` makes sense for the ticker constants, but the generic type aliases still make sense in `providertypes` to me

Reply:
going to add
```
	// PriceResponse is a type alias for the response from the price providers.
	PriceResponse = providertypes.GetResponse[mmtypes.Ticker, *big.Int]
```